Sources/BrowserServicesKit/FeatureFlagger/FeatureFlagger.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  FeatureFlagger.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public protocol FeatureFlagger {
+
+/// Called from app features to determine whether a given feature is enabled.
+///
+/// `forProvider: F` takes a FeatureFlag type defined by the respective app which defines from what source it should be toggled
+/// see `FeatureFlagSourceProviding` comments below for more details
+    func isFeatureOn<F: FeatureFlagSourceProviding>(forProvider: F) -> Bool
+ }
```

Comment:
Stray whitespace snuck in here:

```suggestion
}
```